File: assignments/assignment-2-solution/src-csharp/TeaShop/UserInterface/PaymentBuilder/PaymentBuilderListFactory.cs
namespace Assignment2Solution.UserInterface.PaymentBuilder;

/// <summary>
///     Provides a factory for creating a predefined list of payment strategy generators.
///     This class is responsible for returning a read-only list of objects implementing
///     the <see cref="IPaymentBuilder" /> interface, representing various payment
///     methods that can be used to generate payment strategies.
/// </summary>
public static class PaymentBuilderListFactory
{
    public static IReadOnlyList<IPaymentBuilder> Get()
    {
        return new List<IPaymentBuilder>
        {
            new CreditCardPaymentBuilder(),
            new ApplePayPaymentBuilder(),
            new CryptoCurrencyPaymentBuilder()
        }.AsReadOnly();
    }
}

File: assignments/assignment-2-solution/src-java/src/main/java/assignment2solution/userinterface/paymentbuilder/PaymentBuilderListFactory.java
package assignment2solution.userinterface.paymentbuilder;

import java.util.List;

/**
 * Provides a factory for creating a predefined list of payment strategy generators.
 */
public final class PaymentBuilderListFactory {
    private PaymentBuilderListFactory() {
    }

    public static List<IPaymentBuilder> get() {
        return List.of(
            new CreditCardPaymentBuilder(),
            new ApplePayPaymentBuilder(),
            new CryptoCurrencyPaymentBuilder()
        );
    }
}
